src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/PropertiesShouldNotBeWriteOnlyTests.cs

```diff
@@ -19,5 +19,673 @@ protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_protected_Read_private_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class ClassWithReadableProperty
+    {
+        protected string field;
+        protected string AccessibleProperty2
+        {
+        get { return field; }
+        private set { field = value; }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_internal_Read_private_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        internal string AccessibleProperty3
+        {
+            get { return field; }
+            private set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_protected_internal_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        protected internal string AccessibleProperty4
+        {
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        public string AccessibleProperty5
+        {
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_Read_protected_Write()
+        {
+            var code = @"
+using System;
+namespace DesignLibrary
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        public string AccessibleProperty6
+        {
+            get { return field; }
+            protected set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_override_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    protected string field;
+    public class DerivedClassWithReadableProperty : ClassWithReadableProperty
+    {
+        public override string ReadableProperty1
+        {
+            set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+// Cannot get below test to work, not sure where the issue is.
+//        [Fact]
```

Comment:
According to the old fxcop code, these were suppose to be good code.  But I wasn't able to get the expected result.